jbi/runner.py

```diff
@@ -38,23 +39,26 @@ def execute_action(
         )
 
         try:
-            bug_obj: BugzillaBug = request.bugzilla_object
+            if request.bug.is_private:
+                request = request.copy(
```

Comment:
One thing I'm concerned with here: while it's more convenient to set `request.bug` as the result of `get_bugzilla().get_bug`, I'm wondering if that will muddle our logging if over the course of an action run, we modify the request. (e.g., when we received the request it was a private bug, but now it's the fetched bug with all of the data. How did that happen 🤔)

Maybe we should modify `__call__` signatures to that they accept `bug` and `event` as parameters. Then we can do:

```python
bug = request.bug
if request.bug.is_private:
    bug = get_bugzilla().get_bug(bug.id)
handled, details = action.caller(bug=bug, event=request.event)
```

Or maybe that's logic we encourage action authors to handle _in_ call:

```python
def __call__(self, payload):
    bug, event = payload.bug, payload.event
    if bug.is_private:
        bug = get_bugzilla.get_bug(bug.id)
    ...
```

but in both cases, it feels to me like the request should be treated as immutable. 
What do you think?

Reply:
I agree in the long term modifying the request could be confusing.

I think between the suggestions, I prefer the latter; providing the 'payload' or a 'data' object to the action.

(When we discuss composition of actions, the generic 'data' object could be part of a consistent method_signature for internal tasks that could be ordered and enabled, breaking Actions down into smaller tasks?)

But also, not sure if this should be tackled in this PR.

Reply:
I agree, removing mutations of the original request is better.

I kind of prefer passing the `bug` and `event` objects to the actions, instead of duplicating the extraction in the actions. Even if that means we potentially have an additional fetch for actions that may not require it.

If you don't mind, I'd like to address this in a follow-up, so that we can ship #224 and #217 first

Reply:
Can we open an issue to track this?

Reply:
https://github.com/mozilla/jira-bugzilla-integration/issues/227